### ISSUE TITLE
Static cache takes up too much memory

### ISSUE THREAD

**Issue by lindexi** [OfficeDev/Open-XML-SDK#1062]
# Before submitting an issue, please fill this out

Is this a:
- [ ] Issue with the OpenXml library
- [ ] Question on library usage

If you have answered that this is a question, please ask it on StackOverflow instead of here.
This issue tracker is meant to track product issues while StackOverflow excels at answering questions

---------------- Remove this line and above before posting ----------------

**Description**

I capture my application by dot Memory tool. And I find the OpenXML SDK take up too many memories to cache.

![](https://user-images.githubusercontent.com/16054566/140276905-2f298141-b624-4a95-9f98-201d780894c2.png)

My application will use the OpenXML SDK to read the file once, and it means that the cache in SDK is wasteful

**Information**

- .NET Target: ALL
- DocumentFormat.OpenXml Version: 2.13.0

**Repro**

Not special

**Observed**

The OpenXML SDK take up too many memories to cache.

**Expected**

We can free it manual.

**Comment by twsouthwick** [comment]
Agreed! I'm working on removing the static aspect of it so the data is something that you can control yourself. That's part of the goal with the `Features` that will allow for controlling that manually. Probably not till v3, but I'm moving towards that.

The majority of the data is currently held in the `DefaultFeatureCollection` object that is created once and shared among all usages. We could probably move it into a `WeakReference<>` so that it's ejected from memory if no one is using it. The only concern is if people are processing things serially, it would end up being recreated a lot. Maybe for now it could be something that is hidden behind an AppContext switch so you could opt into the WeakReference aspect if you want.

Would you be interested in providing a PR for that?

**Comment by twsouthwick** [comment]
As an aside, you can also set things up in an unloadable `AssemblyLoadContext` if you are only doing it once and want *all* the infrastructure to be removed from memory (i.e. types, jitted code, statics, etc). If you're still on .NET Framework, then you can use an `AppDomain` for similar isolation.

**Comment by lindexi** [comment]
@twsouthwick  Thank you very much.

**Comment by twsouthwick** [comment]
@lindexi are you interested in adding a fix for this?

**Comment by lindexi** [comment]
I'm too busy recently. I'm afraid I don't have time to do it. Thank you.
